src/pydap/tests/test_cas_urs.py

```diff
@@ -0,0 +1,50 @@
+from pydap.client import open_url
+from pydap.cas import urs
+import pydap.net
+import requests
+import os
+import sys
+import logging
+if sys.version_info < (2, 7):
+    import unittest2 as unittest
+else:
+    import unittest
+
+logging.basicConfig(filename='cas_urs_test.log', level=logging.DEBUG)
+
+
+class TestUrs(unittest.TestCase):
+    url = ('https://goldsmr3.gesdisc.eosdis.nasa.gov/opendap/'
+           'MERRA_MONTHLY/MAIMCPASM.5.2.0/1979/'
+           'MERRA100.prod.assim.instM_3d_asm_Cp.197901.hdf')
+
+    def test_basic_urs_auth(self):
+        """Set up PyDAP to use the URS request() function"""
+        session = urs.setup_session(os.environ['USERNAME_URS'],
+                                    os.environ['PASSWORD_URS'],
+                                    check_url=self.url)
+
+        test_url = self.url + '.dods?SLP[0:1:0][0:1:10][0:1:10]'
+        res = requests.get(test_url, cookies=session.cookies)
```

Comment:
I've done exactly as you recommended. ``resquests`` is now required in both cas_extras and testing_extras.